apps/req.c

```diff
@@ -696,17 +696,21 @@ int req_main(int argc, char **argv)
 
         EVP_PKEY_CTX_free(genctx);
         genctx = NULL;
+    }
```

Comment:
Could you please separate changes in this file to a separate commit? With appropriate commit message, of course. It is not clear what this is fixing/changing.

Reply:
Ah, yes, this should have been more explicit.
The point is that, so far, the `-keyout` option was ignored unless a new key is created.
With the change, which is now in a separate commit, the `keyout` option respected also in case the `-key` option is given.
This is needed by the improved tests that no more use the key generation feature of `req` 
but a pre-computed key from the file now given using the `key` option.
On this occasion I also improved the documentation of the `req` options `-new`, `-newkey`, and `-keyout`.